src/harrastuspassi/harrastuspassi/api.py

```diff
@@ -75,10 +78,62 @@ def filter(self, qs, value):
         return super().filter(qs, list(values_with_children))
 
 
+class NearestOrderingFilter(filters.OrderingFilter):
+    def get_coordinates(self):
+        errors = defaultdict(list)
+        try:
+            near_latitude = self.parent.data.get('near_latitude')
+        except KeyError:
+            errors['near_latitude'].append(_('This field is required when nearest ordering is used.'))
+        try:
+            near_longitude = self.parent.data.get('near_longitude')
+        except KeyError:
+            errors['near_longitude'].append(_('This field is required when nearest ordering is used.'))
+        try:
+            near_latitude = float(near_latitude)
+        except ValueError:
+            errors['near_latitude'].append(_('Must be a float.'))
+        try:
+            near_longitude = float(near_longitude)
+        except ValueError:
+            errors['near_longitude'].append(_('Must be a float.'))
+        try:
+            assert near_latitude >= -180.0
+            assert near_latitude <= 180.0
+        except AssertionError:
+            errors['near_latitude'].append(_('Value must be within -180.0 and 180.0.'))
+        try:
+            assert near_longitude >= -90.0
+            assert near_longitude <= 90.0
+        except AssertionError:
+            errors['near_longitude'].append(_('Value must be within -90.0 and 90.0.'))
+        if errors:
+            raise ValidationError(errors)
+        return near_latitude, near_longitude
+
+    def filter(self, qs, value):
+        if value in EMPTY_VALUES:
+            return qs
+        ordering = [self.get_ordering_value(param) for param in value]
+        if 'distance_to_point' in ordering or '-distance_to_point' in ordering:
+            near_latitude, near_longitude = self.get_coordinates()
+            qs = qs.order_by_distance_to(near_latitude, near_longitude)
+        return qs.order_by(*ordering)
```

Comment:
I think `qs.order_by_distance_to()` returns a queryset with ordering, then this line overwrites what it did? 

> Each order_by() call will clear any previous ordering.

Reply:
This was by design, but admittedly confusing. Tweaked a bit.